source/read/retrieve.txt

```diff
@@ -0,0 +1,108 @@
+.. _kotlin-sync-retrieve:
+
+=============
+Retrieve Data
+=============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+ 
+.. meta::
+   :keywords: code examples, read, search, cursor
+
+Overview
+--------
+
+In this guide, you can learn how to use the {+driver-long+} to retrieve data from a
+MongoDB collection by using read operations. You can call the ``find()`` method to
+retrieve documents that match a set of criteria.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``restaurants`` collection in the ``sample_restaurants``
+database from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see the
+:atlas:`Get Started with Atlas </getting-started>` guide.
+
+.. _kotlin-sync-retrieve-find:
+
+Find Documents
+--------------
+
+The {+driver-short+}'s ``find()`` method retrieves documents from a collection. This
+method takes a **query filter** and returns all matching documents. A query filter is an
+object that specifies the documents you want to retrieve in your query.
+
+.. To learn more about query filters, see :ref:`kotlin-sync-specify-query`.
+
+Find Documents Example
+~~~~~~~~~~~~~~~~~~~~~~
+
+The following example uses the ``find()`` method to find all documents where the
+``cuisine`` field has the value ``Spanish``:
+
+.. code-block:: kotlin
+   :copyable: true
+
+   val results = collection.find(eq("cuisine", "Spanish"))
+
+The ``find()`` operation in the preceding example returns a ``FindIterable`` object,
+which can be iterated over by using the ``forEach()`` method, as shown in the following
+example:
+
+.. code-block:: kotlin
+   :copyable: true
+
+   val results = collection.find(eq("cuisine", "Spanish"))
+   results.forEach { result ->
+     println(result)
+   }
+
+To find all documents in a collection, pass an empty filter to the ``find()`` method:
+
+.. code-block:: kotlin
+    :copyable: true
+
+    val allRestaurants = collection.find()
+
+Modify Find Behavior
+~~~~~~~~~~~~~~~~~~~~
+
+You can modify the behavior of the ``find()`` method by chaining methods to
+the ``find()`` method call.
+
+The following example uses additional methods to limit the number of documents returned
```

Comment:
```suggestion
The following example chains the ``limit()`` and ``maxTime()`` methods to limit the number of documents returned
```

Reply:
In general, please resolve vale issues before sending PR for review!

Reply:
Good callout, that's on me for missing that